Failing test: Security Solution Cypress.x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/page_filters/changed_banner·cy·ts - Alerts page filters - changed banner "before each" hook for "should populate custom filters & display the changed banner" "before each" hook for "should populate custom filters & display the changed banner"

A test failed on a tracked branch

```
AssertionError: Timed out retrying after 150000ms: Expected <span.euiLoadingSpinner.css-1fsqg4c-euiLoadingSpinner-l> not to exist in the DOM, but it was continuously found.

Because this error occurred during a `before each` hook we are skipping the remaining tests in the current suite: `Alerts page filters - chang...`
    at waitForAlerts (webpack:///./tasks/alerts.ts:454:28)
    at Context.eval (webpack:///./e2e/investigations/alerts/page_filters/changed_banner.cy.ts:64:18)
```

First failure: [kibana-on-merge - 9.3](https://buildkite.com/elastic/kibana-on-merge/builds/91359#019d0abc-1eb2-4f74-a3ce-f8e66ea4149c)

<!-- kibanaCiData = {"failed-test":{"test.class":"Security Solution Cypress.x-pack/solutions/security/test/security_solution_cypress/cypress/e2e/investigations/alerts/page_filters/changed_banner·cy·ts","test.name":"Alerts page filters - changed banner \"before each\" hook for \"should populate custom filters & display the changed banner\" \"before each\" hook for \"should populate custom filters & display the changed banner\"","test.failCount":1}} -->

--- a/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/alerts.ts
+++ b/x-pack/solutions/security/test/security_solution_cypress/cypress/tasks/alerts.ts
@@ -465,7 +465,11 @@ export const showTopNAlertProperty = (propertySelector: string, rowIndex: number
 export const waitForAlerts = () => {
   waitForPageFilters();
   cy.get(REFRESH_BUTTON).should('not.have.attr', 'aria-label', 'Needs updating');
-  cy.get(DATAGRID_CHANGES_IN_PROGRESS).should('not.be.true');
+  // `.should('not.be.true')` on a jQuery collection is always true (a jQuery object is
+  // never strictly `=== true`), so this used to be a silent no-op regardless of whether
+  // the data grid was mid-refresh. Assert on absence of the element instead so Cypress
+  // actually retries until the in-progress indicator is gone.
+  cy.get(DATAGRID_CHANGES_IN_PROGRESS).should('not.exist');
   cy.get(EVENT_CONTAINER_TABLE_LOADING).should('not.exist');
   cy.waitForNetworkIdle('/internal/search/privateRuleRegistryAlertsSearchStrategy', 500);
 };

